### PR TITLE
Tooltip support in the Select component

### LINKED ISSUE #1495: Add built-in tooltips to the Select options
**Is your feature request related to a problem?**
Recently, we have been requested to add support to longer options labels in the Select component. Since implementing a line break would lead to ugly scenarios (visually) and the list should have the same size as the input, we have decided to use this HTML solution to display the full label without the need to address some unexpected scenarios inside the list.

**Describe the solution you'd like**
Add the `title` prop to the Select options.

Sin título

**Additional context**
The full discussion can be founded [here](https://github.com/dxc-technology/halstack-react/discussions/901)

### PR DESCRIPTION
**Pull request by GomezIvann**
**Checklist**
- [x] The build process is done without errors. All tests pass in the `/lib` directory.
- [x] Self-reviewed the code before submitting.
- [x] Meets accessibility standards.
- [x] Added/updated documentation to `/website` as needed.
- [ ] Added/updated tests as needed.

**Description**
A tooltip has been added to the Select component. When the label of an option, both when it is selected and in the listbox, is too long, a built-in tooltip will appear to help the user see the full text.

This functionality should be tested manually since there's no way of testing refs and document object calls with React Testing Library or snapshot testing (chromatic).

**Closes #1495**

### CODE REVIEW

File: lib/src/select/Option.tsx [hunk 1]
```diff
@@ -13,41 +13,49 @@ const Option = ({
...
-    selected={isSelected}
-    role="option"
-    aria-selected={isSelected}
-  >
-    <StyledOption
+}: OptionProps): JSX.Element => {
+  const labelRef = useRef<HTMLSpanElement>(null);
+
+  useEffect(() => {
+    if (labelRef.current.scrollWidth > labelRef.current.clientWidth) document.getElementById(id).title = option.label;
```
Comment (Jialecl) [inline lib/src/select/Option.tsx:20]: The first time opening a select labelRef.current.clientWidth can be 0.

**Comment by Jialecl** [comment]
It seems that the tooltip is not updated correctly whenever options are deselected.
image

**Comment by Jialecl** [comment]
The font color of the selected values seems to change if the select has a placeholder and the list is opened.
image

File: lib/src/select/Option.tsx (hunk 1 above)
Comment (GomezIvann) [reply lib/src/select/Option.tsx:20]: The `useEffect` executes after the component is rendered, how could that be possible?

Comment (Jialecl) [reply lib/src/select/Option.tsx:20]: image
image
These are screenshot from the first opening.

image
And this one after closing and reopening the list.

**Review by Jialecl — APPROVED** [review]

**Pull request merged**